napari_skimage_regionprops/_regionprops.py

```diff
@@ -184,3 +196,371 @@ def ellipsoid_axis_lengths(table):
+    """Return an element-wise dictionary of two lists.
+
+    From a list of keys and a list of values, return a dictionnary linking
+    those lists. Both lists must have the same length.
+
+    Parameters
+    ----------
+    list_of_keys : List
+        list to be used as dictionnary keys.
+    list_of_values : List
+        list to be used as dictionnary values.
+
+    Returns
+    -------
+    Dictionnary
+    """
+    return dict(
+                list(
+                    map(
+                        lambda list_of_keys, list_of_values:
+                        [list_of_keys, list_of_values],
+                        *[list_of_keys, list_of_values]
+                        )
+                    )
+                )
+
+
+@register_function(
+    menu="Measurement > Regionprops map multichannel (scikit-image, nsr)")
+def napari_regionprops_map_channels_table(
+        label_images: List[napari.types.LabelsData],
+        intensity_images: List[napari.types.ImageData],
+        reference_label_image: napari.types.LabelsData,
+        intersection_area_over_object_area: float = 0.5,
+        return_summary_statistics: bool = True,
+        size: bool = True,
+        intensity: bool = True,
+        perimeter: bool = False,
+        shape: bool = False,
+        position: bool = False,
+        moments: bool = False,
+        napari_viewer: Viewer = None) -> "pandas.DataFrame":
+    """
+    Add a table widget to a napari viewer with mapped summary statistics.
+
+    Adds a table widget to a given napari viewer with summary statistics that
+    relates objects of one reference label image to objects in other label
+    images. Matched intensity images should be given for intensity
+    measurements. If no intensity images are given, calculates only properties
+    not related to intensities. If a single label image is given, it executes
+    regular 'regionprops_table' function.
+    """
+    multichannel = True
+    suffixes = []
+    if napari_viewer is not None:
+        # store list of labels layers for saving results later
+        labels_layer_list = [None]*len(label_images)
+        for layer in napari_viewer.layers:
+            if type(layer) is napari.layers.Labels:
+                # Store in the same order as label_images
+                for i, labels in enumerate(label_images):
+                    if np.array_equal(layer.data, labels):
+                        labels_layer_list[i] = layer
+                        suffixes += [layer.name]
+                    # Store reference channel index
+                    if np.array_equal(reference_label_image, labels):
+                        ref_channel = i
+    # If single label image is provided, indicate to do single channel regionprops
+    # ref_channel is used as a flag to indicate single or multi-channel
+    if len(label_images) == 1:
+        ref_channel = None
+        multichannel = False
+        label_images = label_images[0]
```

Comment:
I'm not sure if I understand why a differentiation between number of channels `n > 1` is necessary. Would it make the functionality/code easier/shorter if analysing multiple channels with `n=1` was possible? I would like to avoid a user guide saying "if you provide this, a result A will be produced. If you provide something else, something else will be done"

Reply:
This was removed now.
It was a way to identify how many channels were provided.